### PR TITLE
API-1835: tremendously increase QPS for manifest client to avoid delay

### PR DESCRIPTION
**Pull request by deads2k**
found while testing authentication operator.

### CODE REVIEW

File: pkg/manifestclient/readwrite_roundtripper.go [hunk 1]
```diff
@@ -5,11 +5,20 @@ import (
...
 	"io/fs"
+	"k8s.io/client-go/rest"
 	"net/http"
 	"os"
 )
 
-// Enter here and call `NewForConfigAndClient(&rest.Config{}, httpClient)`
+// RecommendedRESTConfig is meant to be paired with the HTTPClients below
+func RecommendedRESTConfig() *rest.Config {
+	return &rest.Config{
```
Comment (p0lyn0mial) [inline pkg/manifestclient/readwrite_roundtripper.go:15]: Is this configuration required for the fake client, the live client, or both? 
Why can't we use the configuration that is already used by the operator? (IIRC provided by the library-go)

Comment (deads2k) [reply pkg/manifestclient/readwrite_roundtripper.go:15]: > Is this configuration required for the fake client, the live client, or both?
> Why can't we use the configuration that is already used by the operator? (IIRC provided by the library-go)

when wiring to actual operators, it turns out we don't take the input that way.  This is a very natural package and direct peer to the http client that's wired in usage.

Comment (p0lyn0mial) [reply pkg/manifestclient/readwrite_roundtripper.go:15]: I've noticed that the live client [will use ](https://github.com/openshift/cluster-authentication-operator/pull/718/files#diff-a58d662aec11fcd8b98a7666c97bc4de17040abbfed4535d56055b40b78fc2e5R19)the same settings. 
Currently, the QPS settings are configured by the [controllercmd](https://github.com/openshift/library-go/blob/2ba740733b461ee8fe7e95a7d95f06b30ebcaa1d/pkg/controller/controllercmd/builder.go#L428). 

Is there a reason we can't preserve the current settings only for the live client?

Comment (deads2k) [reply pkg/manifestclient/readwrite_roundtripper.go:15]: like cloud providers, external OIDC providers.  We'll probably (definitely) discover that direct kube-apiserver checks actually fail in the general case

**Comment by deads2k** [comment]
We can re-open the discussion on monday, but I'm trying to resolve the all the vendored deps for our first integration test in auth-operator and this makes us go past our timeout.

Basically, because this rest.Config has no connection information in it, you cannot use it to connect somewhere useful.

**Pull request merged**